### PR TITLE
Disable swap space on nodes at install and upgrade

### PR DESCRIPTION
**Pull request by mtnbikenc**
https://trello.com/c/vGmZYJ79/296-3-disable-swap-at-install-and-upgrade

### CODE REVIEW

**Comment by mtnbikenc** [comment]
aos-ci-test

**Comment by sdodson** [comment]
How much of a challenge do you think it'd be to add a task that checks to ensure that we have enough free memory versus swap in use before doing this and perform that as a pre-upgrade check? I believe the formula we'd use is to grab MemFree, SwapTotal, SwapFree from /proc/meminfo and abort if `(MemFree < (SwapTotal - SwapFree) )`

The output of `free` may be easier too.

**Comment by sdodson** [comment]
That said, if we disable swap while the node is evacuated the memory usage will be vastly different than what it is when prior to performing the upgrade and it's difficult to determine how evacuation will affect usage.

File: roles/openshift_node/tasks/main.yml [hunk 1]
```diff
@@ -34,6 +34,32 @@
         dns_ip: "{{ openshift_dns_ip | default(none) | get_dns_ip(hostvars[inventory_hostname])}}"
         env_vars: "{{ openshift_node_env_vars | default(None) }}"
 
+# https://docs.openshift.com/container-platform/3.4/admin_guide/overcommit.html#disabling-swap-memory
+- name: Check for swap usage
+  command: grep swap /etc/fstab
```
Comment (sdodson) [inline roles/openshift_node/tasks/main.yml]: `grep -v "^#" /etc/fstab | grep swap` or something better to avoid false positives on commented lines?

**Comment by mtnbikenc** [comment]
Updated grep command to ignore commented lines.

**Comment by mtnbikenc** [comment]
aos-ci-test

**Comment by sdodson** [comment]
aos-ci-test

**Comment by mtnbikenc** [comment]
aos-ci-test

**Comment by mtnbikenc** [comment]
[merge]

**Comment by mtnbikenc** [comment]
aos-ci-test

**Comment by mtnbikenc** [comment]
Flake https://github.com/openshift/origin/issues/13271

**Comment by mtnbikenc** [comment]
[merge]

**Comment by mtnbikenc** [comment]
Flake https://github.com/openshift/origin/issues/12629 Downward API volume
Flake https://github.com/openshift/origin/issues/13705 Dumping container logs
re[merge]

**Comment by rhcarvalho** [comment]
> How much of a challenge do you think it'd be to add a task that checks to ensure that we have enough free memory versus swap in use before doing this and perform that as a pre-upgrade check? I believe the formula we'd use is to grab MemFree, SwapTotal, SwapFree from /proc/meminfo and abort if (MemFree < (SwapTotal - SwapFree) )
> 
> The output of free may be easier too.

@mtnbikenc @sdodson the builtin Ansible "setup" module provides memory-related facts such that we may not need to call `free` ourselves. We've been using it in https://github.com/openshift/openshift-ansible/pull/3518.
In particular, we look at `ansible_memtotal_mb`:
https://github.com/openshift/openshift-ansible/pull/3518/files#diff-7cb3c3806a4642e46d907eeed0c8873eR28

**Comment by sdodson** [comment]
So, based on that if we choose to add it, if ( ansible_memfree_mb - ansible_memory_mb.swap.used > 0 ) we're good to go? Anyway, given we're performing this either at install time or during upgrade when the node is already drained I'm relatively confident that we won't run into problems.

```
        "ansible_memory_mb": {
            "nocache": {
                "free": 3507, 
                "used": 284
            }, 
            "real": {
                "free": 2012, 
                "total": 3791, 
                "used": 1779
            }, 
            "swap": {
                "cached": 0, 
                "free": 0, 
                "total": 0, 
                "used": 0
            }
        }, 
        "ansible_memtotal_mb": 3791, 
```

**Pull request merged**